plutus-chain-index/app/Marconi/Index/Utxo.hs

```diff
@@ -114,7 +113,14 @@ query
   -> IO Result
 query ix addr updates = do
   -- SELECT all utxos that have not been spent.
-  storedUtxos <- SQL.query (ix ^. Ix.handle) "SELECT address, txId, inputIx FROM utxos LEFT JOIN spent ON utxos.txId = spent.txId AND utxos.inputIx = spent.inputIx WHERE utxos.txId IS NULL AND utxos.address = ?" (Only addr)
+  let c = ix ^. Ix.handle
+  -- Create indexes initially. When created this should be a no-op.
```

Comment:
Doing it a query time is a clever idea. The user can decide how much it wants to wait before creating the indices. 

How is  SQLite going to handle creating the indexes while inserting new data? is it going to block the writes or can it do it concurrently?

Reply:
I think it does the right thing (tm), but I have not tested it.